cosi.proto

```diff
@@ -0,0 +1,183 @@
+    rpc ProvisionerGrantBucketAccess (ProvisionerGrantBucketAccessRequest) returns (ProvisionerGrantBucketAccessResponse);
+    // This call revokes all access to a particular bucket from a principal.
+    rpc ProvisionerRevokeBucketAccess (ProvisionerRevokeBucketAccessRequest) returns (ProvisionerRevokeBucketAccessResponse);
+}
+
+message ProvisionerGetInfoRequest {
+    // Intentionally left blank
+}
+
+message ProvisionerGetInfoResponse {    
+    // This field is REQUIRED
+    // The name MUST follow domain name notation format
+    // (https://tools.ietf.org/html/rfc1035#section-2.3.1). It SHOULD
+    // include the plugin's host company name and the plugin name,
+    // to minimize the possibility of collisions. It MUST be 63
+    // characters or less, beginning and ending with an alphanumeric
+    // character ([a-z0-9A-Z]) with dashes (-), dots (.), and
+    // alphanumerics between.
+    string name = 1;
+}
+
+message ProvisionerCreateBucketRequest {    
+    // This field is REQUIRED
+    // Bucket name.
+    string bucket_name = 1;
+
+    // This field is OPTIONAL
+    // Protocol specific information required by the call is passed in as key,value pairs.
+    map<string,string> bucket_context = 2;
+
+    enum AnonymousBucketAccessMode {
+        // Default, disallow uncredentialed access to the backend storage.
+	BUCKET_PRIVATE = 0;
+        // Read only, uncredentialed users can call ListBucket and GetObject.
+	BUCKET_READ_ONLY = 1;
+        // Write only, uncredentialed users can only call PutObject.
+	BUCKET_WRITE_ONLY = 2;
+        // Read/Write, same as ro with the addition of PutObject being allowed.
+	BUCKET_READ_WRITE = 3;
+    }
+    
+    // This field is OPTIONAL
+    // Allow uncredentialed access to bucket.
+    AnonymousBucketAccessMode anonymous_bucket_access_mode = 3;
+}
+
+message ProvisionerCreateBucketResponse {
+    // Intentionally left blank
+}
+
+message ProvisionerDeleteBucketRequest {
+    // This field is REQUIRED
+    // Bucket name.
+    string bucket_name = 1;
+    
+    // This field is OPTIONAL
+    // Protocol specific information required by the call is passed in as key,value pairs.
+    map<string,string> bucket_context = 2;
+}
+
+message ProvisionerDeleteBucketResponse {
+     // Intentionally left blank
+}
+
+message ProvisionerGrantBucketAccessRequest {
+    // This field is REQUIRED
+    // Bucket name.
+    string bucket_name = 1;
+    
+    // This field is OPTIONAL
+    // Protocol specific information required by the call is passed in as key,value pairs.
+    map<string,string> bucket_context = 2;
+
+    // This field is OPTIONAL
+    // Principal for which access is requested
+    string principal = 3;
+    
+    // This field is REQUIRED
+    // Requested Access policy, ex: {"Effect":"Allow","Action":"s3:PutObject","Resource":"arn:aws:s3:::profilepics/*"}
+    string access_policy = 4;
+}
+
+message ProvisionerGrantBucketAccessResponse {
```

Comment:
Can you specify which field is required/optional for this message too?

Reply:
fixed the field description.